File: RemoveDuplicatesFromList/oneloop.c
if (head) {
    struct ListNode *p = head;
    while (p->next) {
        if (p->val != p->next->val) {
            p = p->next;
        }
        else {
            struct ListNode *tmp = p->next;
            p->next = p->next->next;
            free(tmp);
        }
    }
}

return head;

File: RemoveDuplicatesFromList/oneloop.java
public ListNode deleteDuplicates(ListNode head) {
    if(head == null) return null;

    ListNode notDuplic = head;
    ListNode index = head.next;

    while(index!=null){
        if(notDuplic.val != index.val){
            notDuplic.next = index;
            notDuplic = index;
        }

        index = index.next;
    }

    notDuplic.next =null;

    return head;
}
